### PR TITLE
Custom Auth Front

### PR DESCRIPTION
**Pull request by KINFOO**
## Changes proposed

- Change formatting to [Prettier](https://prettier.io)
- A `authFrontServer` parameter to `setup.json`, a specific *Auth-Front* enables to create systems in an Airvantage running locally

### Local

```bash
$ vleet
 Launching Vleet simulation
  ✱ Initializing simulation
   ❙ Configuration file "setup.json"
    ➔ Name: mcs
    ➔ Data center: local
    ➔ User: 🐱
    ➔ Fleet size: 300
    ➔ Simulation Label: dd_lx40_
  ❙ Auth Front "localhost:8787"
   ✔ Authenticated
   ✔ Application already exists
```

### Distant

```sh
$ vleet 
 Launching Vleet simulation
  ✱ Initializing simulation
   ❙ Configuration file "setup.json"
    ➔ Name: mcs
    ➔ Data center: eu
    ➔ User: 🐱
    ➔ Fleet size: 300
    ➔ Simulation Label: PLTFRS-12711
   ✔ Authenticated
  ✱ Creating virtual application: PLTFRS-12711 Dreaming about TJ lx40 firmware 1.0
   ✔ Application created with uid: fefa2f9f8d1f425d82bbfbb6e7fd2b28
   ✔ All systems were created
  ✱ Start simulation -  PLTFRS-12711
  ❙ For  100 systems
  ✱ Generate data for system: TJ LX40 #237
  ❙ Minutes offset by day: 1440
   ✱ Day #0
   ❙ Nth value of day: 0 Time : Thursday, October 15th 2020, 12:00:00 am
     _FIRMWARE_VERSION ➔ 4.11.1.F006
     _FIRMWARE_TYPE ➔ ALEOS.PROD.1f
  ✱ Generate data for system: TJ LX40 #279
  ❙ Minutes offset by day: 1440
   ✱ Day #0
   ❙ Nth value of day: 0 Time : Thursday, October 15th 2020, 12:00:00 am
     _FIRMWARE_VERSION ➔ 4.11.1.F006
     _FIRMWARE_TYPE ➔ ALEOS.PROD.1f
  ✱ Generate data for system: TJ LX40 #299
```

<p align="center"></p>

### CODE REVIEW

**Comment by KINFOO** [comment]
I would like to add @asc-it as reviewer, looks like it is not possible for this project

**Comment by sebz** [comment]
> I would like to add @asc-it as reviewer, looks like it is not possible for this project

Just added him as reviewer

File: cli.js [hunk 1]
```diff
@@ -77,24 +76,37 @@ function getSimulationLabel() {
     return [simulation.label];
 }
 
-airvantage.authenticate()
-    .then(function() {
+return new Promise((resolve, reject) => {
+    if (!!serverConfig.authFrontServer) {
```
Comment (sebz) [inline cli.js:80]: Can't you use `const isLocal`?

**Review by sebz — CHANGES_REQUESTED** [review]
One minor remark. 
Also, your smoke test only test the new local version. Did you test it is still working with the prod servers?

**Comment by KINFOO** [comment]
> One minor remark.
> Also, your smoke test only test the new local version. Did you test it is still working with the prod servers?

Just updated __smoke test__ section

**Review by sebz — APPROVED** [review]
LGTM

**Review by asc-it — APPROVED** [review]
LGTM
Vleet runs locally <3

**Pull request merged**